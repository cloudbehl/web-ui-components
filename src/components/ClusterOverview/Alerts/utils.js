import { get } from 'lodash';

export const filterAlerts = alerts => alerts.filter(alert => get(alert, 'labels.alertname') !== 'Watchdog');
