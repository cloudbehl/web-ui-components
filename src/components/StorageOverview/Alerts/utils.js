import { get } from 'lodash';

export const filterAlerts = alerts => alerts.filter(alert => get(alert, 'annotations.storage_type') === 'ceph');
