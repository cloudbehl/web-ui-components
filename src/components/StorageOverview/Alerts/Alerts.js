import React from 'react';
import PropTypes from 'prop-types';

import {
  DashboardCard,
  DashboardCardBody,
  DashboardCardHeader,
  DashboardCardTitle,
} from '../../Dashboard/DashboardCard';
import { StorageOverviewContext } from '../StorageOverviewContext';
import { AlertsBody } from '../../Dashboard/Alert/AlertsBody';
import { filterAlerts } from './utils';

export const Alerts = ({ alertsResponse }) => {
  if (!Array.isArray(alertsResponse)) {
    return false;
  }

  const alerts = filterAlerts(alertsResponse);

  return alerts.length > 0 ? (
    <DashboardCard>
      <DashboardCardHeader>
        <DashboardCardTitle>Alerts</DashboardCardTitle>
      </DashboardCardHeader>
      <DashboardCardBody>
        <AlertsBody alerts={alerts} />
      </DashboardCardBody>
    </DashboardCard>
  ) : (
    false
  );
};

Alerts.propTypes = {
  alertsResponse: PropTypes.array,
};

Alerts.defaultProps = {
  alertsResponse: null,
};

export const AlertsConnected = () => (
  <StorageOverviewContext.Consumer>
    {props => <Alerts alertsResponse={props.alertsResponse} />}
  </StorageOverviewContext.Consumer>
);

AlertsConnected.propTypes = {
  ...Alerts.propTypes,
};

AlertsConnected.defaultProps = {
  ...Alerts.defaultProps,
};
