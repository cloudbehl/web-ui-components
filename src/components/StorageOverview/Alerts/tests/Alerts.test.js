import React from 'react';
import { shallow } from 'enzyme';

import { Alerts } from '../Alerts';
import { default as AlertsFixtures } from '../fixtures/Alerts.fixture';

const testAlerts = () => <Alerts {...AlertsFixtures.props} />;

describe('<Alerts />', () => {
  it('renders correctly', () => {
    const component = shallow(testAlerts());
    expect(component).toMatchSnapshot();
  });
});
