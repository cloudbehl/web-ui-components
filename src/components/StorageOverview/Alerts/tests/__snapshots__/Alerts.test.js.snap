// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Alerts /> renders correctly 1`] = `
<DashboardCard
  className={null}
>
  <DashboardCardHeader
    className={null}
  >
    <DashboardCardTitle>
      Alerts
    </DashboardCardTitle>
  </DashboardCardHeader>
  <DashboardCardBody
    LoadingComponent={[Function]}
    isLoading={false}
  >
    <AlertsBody
      alerts={
        Array [
          Object {
            "annotations": Object {
              "description": "Disk inaccessible on host ip-10-0-154-62.ec2.internal (device xvda)",
              "message": "Disk is inaccessible",
              "storage_type": "ceph",
            },
            "labels": Object {
              "alertname": "CephOSDDiskUnavailable",
              "severity": "critical",
            },
          },
          Object {
            "annotations": Object {
              "description": "Data recovery has been active for over 2h. Contact Support",
              "message": "Data recovery is slow",
              "storage_type": "ceph",
            },
            "labels": Object {
              "alertname": "CephDataRecoveryTakingTooLong",
              "severity": "warning",
            },
          },
        ]
      }
    />
  </DashboardCardBody>
</DashboardCard>
`;
